frontend.java

```diff
@@ -0,0 +1,540 @@
+					if(sCurrentLine == newAccountNumber){
+						return true;
+					}
+				}
+	 
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+		return false;
+	}
+	public static boolean accountAmount(String number){
+		try{
+			int x =Integer.parseInt(number);
+		}catch(NumberFormatException e){
+			return false;
+		}
+		return true;
+	}
+	public static String addZeros(String number, int size){
+		//get the number and subtrack it by 8 and add that many zeros
+		int length = number.length();
+		int additionalZeros = size - length;
+		for( int i = 0 ; i < additionalZeros ; i++ ){
+			number = "0" + number;
+		}
+		return number;
+	}
+	public static boolean lowerLimit(String amount){
+		boolean valid = false;
+		try{
+			int amountInt = Integer.parseInt(amount);
+			if(amountInt > 0){
+				valid = true;
+			}
+		}catch(NumberFormatException e){
+			//should not reach here because we validate before.
+			System.out.println("Error 0001");
+			System.exit(1);
+		}
+		return valid;
+	}
+	public static boolean upperLimit(String amount){
+		boolean valid = false;
+		try{
+			int amountInt = Integer.parseInt(amount);
+			if(frontend.type == 1){
+				//retail mode
+				if(amountInt > 100000){//above 1,000.00
+					valid = false;
+				}else{
+					valid = true;
+				}
+			}else if(frontend.type == 2){
+				//agent
+				if(amountInt > 99999999){//above 999,999.99
+					valid = false;
+				}else{
+					valid = true;
+				}
+			}else{
+				//error
+				//Should never get here b/c in order to perform an action they must be logged in. 
+				System.out.println("Error 0003");
+				System.exit(1);
+			}
+		}catch(NumberFormatException e){
+			//should not reach here because we validate before.
+			System.out.println("Error 0002");
+			System.exit(1);
+		}
+		return valid;
+	}
+	
+	public static boolean withdrawLimit(String accountNumber,
+			String accountAmount) {
+		
+		int totalAmount = Integer.parseInt(accountAmount);
+		for (String transactionStr : frontend.transactionFile){
+			int itemNum = 0;
+			boolean userAccount = false;
+		    for (String str: transactionStr.split("_", 3)){
+		        //loops 3 items
+		    	if(itemNum == 0 ){
+		    		itemNum++;
+		    		userAccount = false;
+		    	}else if(itemNum == 1){
+		    		//account number
+		    		if( str == accountNumber ){
+		    			userAccount = true;
+		    		}
+		    		itemNum++;
+		    	}else{
+		    		//account amount
+		    		if(userAccount == true ){
+		    			int amount = Integer.parseInt(str);
+		    			totalAmount += amount;
+		    			if((totalAmount) > 1000){
+		    				return false;
+		    			}
+		    			userAccount = false;
+		    		}
+		    		itemNum++;
+		    	}
+		    }
+		}
+		return true;
+	}
+	public static void front_withdraw() {
+		//get account number
+		//get amount
+		System.out.print("Account Number:");
+		String accountNumber = frontend.input();
+		if(accountNumber(accountNumber)==false){
+			frontend.error("Error: Invalid account number.");
```

Comment:
Nested if statements makes it hard to read what you are trying to do.  Why not check for the "false" condition and  exit as soon as that's detected.

e.g.:

    public void withdraw() {
        String accountNumber = input();
        String accountName = 
        if (!isValidAccountNumber(accountNumber) || !doesAccountExist(accountName)){
            return;
        } 
        // Do stuff

    }